wolfcrypt/src/random.c

```diff
@@ -1170,41 +1185,89 @@ static int wc_RNG_HealthTestLocal(int reseed)
 #endif
 
     if (reseed) {
-        ret = wc_RNG_HealthTest(1, seedA, sizeof(seedA),
-                                reseedSeedA, sizeof(reseedSeedA),
-                                check, RNG_HEALTH_TEST_CHECK_SIZE);
-        if (ret == 0) {
-            if (ConstantCompare(check, outputA,
-                                RNG_HEALTH_TEST_CHECK_SIZE) != 0)
-                ret = -1;
+#ifdef WOLFSSL_PROGMEM_AVR
+        byte* seedA = (byte*)XMALLOC(sizeof(_seedA), NULL,
+                             DYNAMIC_TYPE_TMP_BUFFER);
+        byte* reseedSeedA = (byte*)XMALLOC(sizeof(_reseedSeedA), NULL,
+                             DYNAMIC_TYPE_TMP_BUFFER);
+        byte* outputA = (byte*)XMALLOC(sizeof(_outputA), NULL,
+                             DYNAMIC_TYPE_TMP_BUFFER);
+
+        if (!seedA || !reseedSeedA || !outputA) {
+            free(seedA);
+            free(reseedSeedA);
+            free(outputA);
+            ret = MEMORY_E;
+        }
+        else {
+
+            memcpy_P(seedA, _seedA, sizeof(_seedA));
```

Comment:
Consider `XMEMCPY`.

Reply:
For the use of `memcpy_P` it is different to memcpy.  It reads from flash memory rather than SRAM, which requires different opcodes on the processor.

Should I define a `MEMCPY_P` which uses XMEMCPY by default and use that instead?

Sorry, just habit for all the others.

Reply:
Then just keep using `memcpy_P` here since it's only the `WOLFSSL_PROGMEM_AVR` code path.